domain-models-runtime/src/test/java/org/folio/rest/persist/PostgresClientIT.java

```diff
@@ -1483,6 +1483,57 @@ public void getByIdAsString(TestContext context) {
     });
   }
 
+  @Test
+  public void getByIdAsStringForUpdate(TestContext context) {
+    Async async1 = context.async();
+    Async async2 = context.async();
+    String id = randomUuid();
+    postgresClient = createFoo(context);
+    postgresClient.save(FOO, id, new StringPojo("x0"), res -> {
+      assertSuccess(context, res);
+      context.assertEquals(id, res.result());
+      Criterion criterion = new Criterion();
+      criterion.addCriterion(new Criteria().addField("'id'").setOperation("=").setVal(id));
+      postgresClient.startTx(conn -> {
+        // lock the row for update
+        postgresClient.getByIdAsStringForUpdate(conn, FOO, id, get -> {
+          // concurrent update will have to wait
+          postgresClient.update(FOO, new StringPojo("x2"), id, rows -> {
+            context.assertTrue(async1.isCompleted());
+            postgresClient.getByIdAsString(FOO, id, get2 -> {
+              assertSuccess(context, get2);
+              context.assertTrue(get2.result().contains("\"x2\""));
+              async2.complete();
+            });
+          });
+          // update first because it holds the lock
+          postgresClient.execute(conn, "SELECT pg_sleep(0.5)", delayRs -> {
+            assertSuccess(context, delayRs);
+            postgresClient.update(conn, FOO, new StringPojo("x1"),
+                new CQLWrapper(criterion), true, rows -> {
+              assertSuccess(context, rows);
+              context.assertFalse(async2.isCompleted());
+              Row row = rows.result().iterator().next();
+              context.assertEquals(id, row.getUUID("id").toString());
+              async1.complete();
+              postgresClient.endTx(conn, done -> {
+                postgresClient.getByIdAsString(FOO, id, get1 -> {
+                  assertSuccess(context, get1);
+                  context.assertTrue(get1.result().contains("\"x1\""), get1.result());
+                });
+              });
+            });
+          });
+          // before update
+          postgresClient.getByIdAsString(FOO, id, get0 -> {
+            assertSuccess(context, get0);
+            context.assertTrue(get0.result().contains("\"x0\""), get0.result());
```

Comment:
This getByIdAsString with assert should be removed, we can assume that save works correctly.